### PR TITLE
JAMES-2836 Metric records for BlobStore Vault

### PR DESCRIPTION
**Pull request by trantienduchn**
vault-1
retention

### CODE REVIEW

**Comment by trantienduchn** [comment]
vault-1

**Comment by trantienduchn** [comment]
It's not done yet, I could record retention metric, will need to investigate

**Comment by Arsnael** [comment]
Why still a draft?

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java [hunk 1]
```diff
@@ -54,14 +55,26 @@
 public class BlobStoreDeletedMessageVault implements DeletedMessageVault {
     private static final Logger LOGGER = LoggerFactory.getLogger(BlobStoreDeletedMessageVault.class);
 
+    private static final String BLOBSTORE_DELETED_MESSAGE_VAULT_METRIC = "deletedMessageVault:blobStore:";
+    static final String APPEND_METRIC_NAME = BLOBSTORE_DELETED_MESSAGE_VAULT_METRIC + "append";
+    static final String LOAD_MIME_MESSAGE_METRIC_NAME = BLOBSTORE_DELETED_MESSAGE_VAULT_METRIC + "loadMimeMessage";
+    static final String SEARCH_METRIC_NAME = BLOBSTORE_DELETED_MESSAGE_VAULT_METRIC + "search";
+    static final String DELETE_METRIC_NAME = BLOBSTORE_DELETED_MESSAGE_VAULT_METRIC + "delete";
+    static final String RETENTION_METRIC_NAME = BLOBSTORE_DELETED_MESSAGE_VAULT_METRIC + "retention";
```
Comment (mbaechler) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java]: I'm still not ok with that `retention` wording (I thought I said it during the related code review). Retention is about keeping things. Here the action is basically to drop everything that we should not retain.
I'd like to not propagate that wording mistake too far so we should refactor now.
What about `clearExpiredMessages`?

Comment (trantienduchn) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java]: okay! I might missed what you said in the past.

**Comment by chibenwa** [comment]
About `DON't review` comments: I reeeeally don't like the PR structure being implicit.

If you have a `don't review` commit please:
 - Use the `[NO_REVIEW]` prefix
 - Give the `#2536` link to the PR corresponding to the changeset
 - Give the JIRA ticket of the corresponding ticket
 - Give the headline of the corresponding PR

If we do not do it:
 - We loose the strategic aspect of the thing: what it exactly depending on, is it needed? That's some topic we need to be able to exchange on, etc...
 - We loose the ability to cooperate on that PR: I don't know what to rebase it against so I won't touch it.

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java]: IMO we have time to do the name update refactoring here.

Comment (trantienduchn) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java]: I renamed it to `deletedExpiredMessages` to make it fit with the method name

File: grafana-reporting/DeletedMessagesVault-1563771591074-dashboard.json [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+            },
+            {
+              "field": "p95",
+              "id": "5",
+              "meta": {},
+              "settings": {},
+              "type": "avg"
+            }
+          ],
+          "query": "name:\"deletedMessageVault:blobStore:retention\"",
```
Comment (mbaechler) [inline grafana-reporting/DeletedMessagesVault-1563771591074-dashboard.json]: retention spotted

**Comment by trantienduchn** [comment]
rebase to resolve conflicts

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
got team approval + one moka approval, it can be merged

**Comment by rouazana** [comment]
:green_apple: before restarting CI

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**